From: Mauro Carvalho Chehab <mchehab@redhat.com>
Date: Tue, 28 Sep 2010 14:34:08 -0400
Subject: [edac] i7300_edac: remove stale get_error_info logic
Message-id: <20100928113408.27e49040@pedra>
Patchwork-id: 28472
O-Subject: [PATCH RHEL5 13/27] i7300_edac: Cleanup: remove get_error_info logic
Bugzilla: 487428
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

As the error logic in this driver came from i5400 driver, it
were using one function to get errors, and another to display.
Let's make it simpler and avoid doing it into two steps.

Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/edac/i7300_edac.c b/drivers/edac/i7300_edac.c
index 3b6da20..3ae8fae 100644
--- a/drivers/edac/i7300_edac.c
+++ b/drivers/edac/i7300_edac.c
@@ -315,10 +315,6 @@ static struct edac_pci_ctl_info *i7300_pci;
  * i7300 Functions related to error detection
  ********************************************/
 
-struct i7300_error_info {
-	int dummy;	/* FIXME */
-};
-
 const char *get_err_from_table(const char *table[], int size, int pos)
 {
 	if (pos >= size)
@@ -331,22 +327,11 @@ const char *get_err_from_table(const char *table[], int size, int pos)
 	get_err_from_table(table, ARRAY_SIZE(table), pos)
 
 /*
- *	i7300_get_error_info	Retrieve the hardware error information from
- *				the hardware and cache it in the 'info'
- *				structure
- */
-static void i7300_get_error_info(struct mem_ctl_info *mci,
-				 struct i7300_error_info *info)
-{
-}
-
-/*
  *	i7300_process_error_global Retrieve the hardware error information from
  *				the hardware and cache it in the 'info'
  *				structure
  */
-static void i7300_process_error_global(struct mem_ctl_info *mci,
-				 struct i7300_error_info *info)
+static void i7300_process_error_global(struct mem_ctl_info *mci)
 {
 	struct i7300_pvt *pvt;
 	u32 errnum, value;
@@ -400,8 +385,7 @@ error_global:
  *				the hardware and cache it in the 'info'
  *				structure
  */
-static void i7300_process_fbd_error(struct mem_ctl_info *mci,
-				    struct i7300_error_info *info)
+static void i7300_process_fbd_error(struct mem_ctl_info *mci)
 {
 	struct i7300_pvt *pvt;
 	u32 errnum, value;
@@ -451,15 +435,14 @@ error_fbd:
 }
 
 /*
- *	i7300_process_error_info Retrieve the hardware error information from
+ *	i7300_check_error Retrieve the hardware error information from
  *				the hardware and cache it in the 'info'
  *				structure
  */
-static void i7300_process_error_info(struct mem_ctl_info *mci,
-				 struct i7300_error_info *info)
+static void i7300_check_error(struct mem_ctl_info *mci)
 {
-	i7300_process_error_global(mci, info);
-	i7300_process_fbd_error(mci, info);
+	i7300_process_error_global(mci);
+	i7300_process_fbd_error(mci);
 };
 
 /*
@@ -501,19 +484,6 @@ static void i7300_clear_error(struct mem_ctl_info *mci)
 }
 
 /*
- *	i7300_check_error	Retrieve and process errors reported by the
- *				hardware. Called by the Core module.
- */
-static void i7300_check_error(struct mem_ctl_info *mci)
-{
-	struct i7300_error_info info;
-	debugf4("MC%d: " __FILE__ ": %s()\n", mci->mc_idx, __func__);
-
-	i7300_get_error_info(mci, &info);
-	i7300_process_error_info(mci, &info);
-}
-
-/*
  *	i7300_enable_error_reporting
  *			Turn on the memory reporting features of the hardware
  */
